### PR TITLE
[chore]: enable len and empty rules from testifylint

### PR DESCRIPTION
**Pull request by mmorel-35**
#### Description

Testifylint is a linter that provides best practices with the use of testify.

This PR enables [len](https://github.com/Antonboom/testifylint?tab=readme-ov-file#len) and [empty](https://github.com/Antonboom/testifylint?tab=readme-ov-file#empty) rules from [testifylint](https://github.com/Antonboom/testifylint)

It also adds testifylint as tool to use with a make command

### CODE REVIEW

**Review by songy23 — COMMENTED** [review]
> This PR enables [len](https://github.com/Antonboom/testifylint?tab=readme-ov-file#len) rule from [testifylint](https://github.com/Antonboom/testifylint)

I don't see `.golangci.yml` being changed, is it missed out?

**Review by songy23 — APPROVED** [review]

File: Makefile.Common [hunk 1]
```diff
@@ -88,3 +91,12 @@ impi: $(IMPI)
...
 	$(GOCMD) mod download
+
+.PHONY: testifylint
+testifylint: $(TESTIFYLINT)
+	@echo "running $(TESTIFYLINT)"
+	@$(TESTIFYLINT) $(TESTIFYLINT_OPT) ./...
+
+.PHONY: testifylint-fix
+testifylint-fix:
+	@$(MAKE) testifylint TESTIFYLINT_OPT="${TESTIFYLINT_OPT} --fix"
```
Comment (bogdandrutu) [inline Makefile.Common:102]: newline please

File: Makefile.Common [hunk 2]
```diff
@@ -88,3 +91,12 @@ impi: $(IMPI)
 .PHONY: moddownload
 moddownload:
 	$(GOCMD) mod download
+
+.PHONY: testifylint
+testifylint: $(TESTIFYLINT)
+	@echo "running $(TESTIFYLINT)"
+	@$(TESTIFYLINT) $(TESTIFYLINT_OPT) ./...
```
Comment (bogdandrutu) [inline Makefile.Common:98]: Is this not covered by `make lint`?

File: Makefile.Common [hunk 3]
```diff
@@ -36,6 +36,9 @@ MULTIMOD     := $(TOOLS_BIN_DIR)/multimod
 PORTO        := $(TOOLS_BIN_DIR)/porto
 SEMCONVGEN   := $(TOOLS_BIN_DIR)/semconvgen
 SEMCONVKIT   := $(TOOLS_BIN_DIR)/semconvkit
+TESTIFYLINT  := $(TOOLS_BIN_DIR)/testifylint
+
+TESTIFYLINT_OPT?= --enable-all --disable=compares,empty,error-is-as,error-nil,expected-actual,float-compare,formatter,go-require,negative-positive,nil-compare,require-error
```
Comment (bogdandrutu) [inline Makefile.Common:41]: Is it possible to read this from golangci?

**Review by iblancasa — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**